WebContent/WEB-INF/jsp/watchList.jsp

```diff
@@ -271,12 +274,20 @@
 
           watchListChanged();
           WatchListDwr.deleteWatchList(deleteId);
-          maybeDisplayDeleteImg();
       }
 
-      function maybeDisplayDeleteImg() {
-          var wlselect = $("watchListSelect");
-          display("watchListDeleteImg", wlselect.options.length > 1);
+      function maybeDisplayDeleteImg(data) {
+          display("watchListDeleteImg", false);
+          if(isOwner(data) || isAdmin) {
+             var wlselect = $("watchListSelect");
+             display("watchListDeleteImg", wlselect.options.length > 0);
+          }
+      }
+
+      function isOwner(data) {
+          return typeof data !== typeof undefined
+          && typeof data.access !== typeof undefined
+          && (data.access == <c:out value="<%= ShareUser.ACCESS_OWNER %>"/>);
```

Comment:
Nit:  _(it is suggestion - can be omitted)_
It can be simplified by using double exclamation marks.  
```javascript
!!data 
```
This code check if data is not null or it is not undefined. So it has the same and more functionality as this line:
```
typeof data !== typeof undefined
```
Example improvement solution:
```javascript
return !!data && !!data.access && (data.access === <c:out value="<%= ShareUser.ACCESS_OWNER %>"/>);
```

But in JavaScript to perform the equality check try to use [strict equality operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) - triple equal marks. ` (===) ` It checks whether its two operands are equal and has the same types.